processor/src/decode.rs

```diff
@@ -0,0 +1,98 @@
+use std::fmt::Display;
+use std::num::ParseIntError;
+use log::{debug, info};
+use vm_core::trace::instruction::{Add, CJmp, Equal, Instruction, Jmp, Mov, Mul, Opcode, Ret, Sub};
+use crate::{FP_REG_INDEX, GoldilocksField};
+
+pub const NO_IMM_INSTRUCTION_LEN: u64 = 1;
+pub const IMM_INSTRUCTION_LEN: u64 = 2;
+
+pub const REG_FIELD_BIT_LEN: u64 = 4;
+pub const REG2_FIELD_BIT_POSITION: u64 = 14;
+pub const REG1_FIELD_BIT_POSITION: u64 = REG2_FIELD_BIT_POSITION+REG_FIELD_BIT_LEN;
+pub const REG0_FIELD_BIT_POSITION: u64 = REG1_FIELD_BIT_POSITION+REG_FIELD_BIT_LEN;
+pub const IMM_FLAG_BIT_LEN: u64 = 1;
+pub const IMM_FLAG_FIELD_BIT_POSITION: u64 = REG0_FIELD_BIT_POSITION+REG_FIELD_BIT_LEN;
+pub const OPCODE_FLAG_FIELD_BIT_POSITION: u64 = IMM_FLAG_FIELD_BIT_POSITION+IMM_FLAG_BIT_LEN;
+
+pub const REG_FIELD_BITS_MASK: u32 = 0xf;
+pub const IMM_FLAG_FIELD_BITS_MASK: u32 = 0x1;
+
+pub fn decode_raw_instruction(raw_inst_str: &str, imm_str: &str) -> Result<(String, u64), ParseIntError>{
+    let mut step = NO_IMM_INSTRUCTION_LEN;
+    let raw_inst = u32::from_str_radix(raw_inst_str.trim_start_matches("0x"), 16)?;
+    debug!("raw_inst:{}", raw_inst);
+    let op_code = Opcode::try_from((raw_inst >> OPCODE_FLAG_FIELD_BIT_POSITION) as u8).unwrap();
```

Comment:
Return error might be better than unwrap.

Reply:
ok, I add an err protection.